src/crewai/task.py

```diff
@@ -326,9 +328,14 @@ def get_agent_by_role(role: str) -> Union["BaseAgent", None]:
 
     def _create_converter(self, *args, **kwargs) -> Converter:
         """Create a converter instance."""
-        converter = self.agent.get_output_converter(*args, **kwargs)
-        if self.converter_cls:
+        if self.agent and not self.converter_cls:
```

Comment:
This was added to fix the Type Checking issue that was before this task.

@joaomdmoura On a second note, I was talking with @bhancockio, and we realized that the "if sef.converter_cls" was added by this PR https://github.com/crewAIInc/crewAI/pull/800

The issue with this overall function is that based on the typed return it should return a Converter, but both self.agent.get_output_converter and self.converter_cls could be None. So for now I'm raising the exception, but do you think is there something better that we can do it here or if its relevant to have this "converter_cls"?